MekHQ/src/mekhq/Utilities.java

```diff
@@ -369,17 +369,29 @@ public static File lastFileModified(String dir, FilenameFilter filter) {
         CampaignOptions options = campaign.getCampaignOptions();
         ArrayList<String> variants = new ArrayList<>();
         for(MechSummary summary : MechSummaryCache.getInstance().getAllMechs()) {
-            // If this isn't the same chassis, is our current unit, or is a different weight we continue
-            if(!en.getChassis().equalsIgnoreCase(summary.getChassis())
+            // If this isn't the same chassis, is our current unit, we continue
+            if (!en.getChassis().equalsIgnoreCase(summary.getChassis())
                     || en.getModel().equalsIgnoreCase(summary.getModel())
-                    || summary.getTons() != en.getWeight()
                     || !summary.getUnitType().equals(UnitType.determineUnitType(en))) {
                 continue;
             }
             // If we only allow canon units and this isn't canon we continue
-            if(!summary.isCanon() && options.allowCanonRefitOnly()) {
+            if (!summary.isCanon() && options.allowCanonRefitOnly()) {
                 continue;
             }
+            
+            // Weight of the two units must match or we continue, but BA weight gets
+            // checked differently
+            if (en instanceof BattleArmor) {
+                if (((BattleArmor)en).getTroopers() != (int) summary.getTWweight()) {
```

Comment:
```suggestion
                if (((BattleArmor) en).getTroopers() != (int) summary.getTWweight()) {
```

Reply:
Also, should this be based on the ADVANCED_TACOPS_BA_WEIGHT option?

Reply:
No, this should be TWweight to work as intended when compared to getTroopers.  I'm pretty sure MechSummary was written before BA/Infantry was fully implemented, as it doesn't have any functionality for tracking the number of troopers.  Infantry/BA refits bear a more comprehensive look at some point, but that's not a rabbithole I want to go down at present.